src/shogun/base/SGObject.h

```diff
@@ -567,6 +567,20 @@ class CSGObject
 			demangled_type<T>().c_str());
 		return nullptr;
 	}
+
+	template <class T> const T* as() const
```

Comment:
you are using this below I assume when changing the signatures of some methods?

Reply:
This is useful for const pointers (`const CFeatures*`). Although we decided to move to const methods later, I leave it here for possible future usage.